File: frontend/src/utils/schemas/klimarelevanzpruefungEingabeFb3.js
import { number, string, object } from 'yup'

// Validation schema
export const schema = object({
  c1q1: number().required('Seite 1, Frage 1'),
  c1q2: number().when('c1q1', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 2'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q3: string().when('c1q2', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 3'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q4: number().when('c1q1', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 4'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q5: string().when('c1q4', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 5'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q6: number().when('c1q1', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 6'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q7: string().when('c1q6', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 7'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q8: number().when('c1q1', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 8'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c1q9: string().when('c1q8', {
    is: 1,
    then: (schema) => schema.required('Seite 1, Frage 9'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q1: number().required('Seite 2, Frage 1'),
  c2q2: number().when('c2q1', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 2').min(0, 'Seite 2, Frage 2: Muss mindestens 0 sein'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q3: string().when('c2q2', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 3'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q4: number().when('c2q1', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 4'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q5: string().when('c2q4', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 5'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q6: number().when('c2q1', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 6'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q7: string().when('c2q6', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 7'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q8: number().when('c2q1', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 8'),
    otherwise: (schema) => schema.nullable(true)
  }),
  c2q9: string().when('c2q8', {
    is: 1,
    then: (schema) => schema.required('Seite 2, Frage 9'),
    otherwise: (schema) => schema.nullable(true)
  })
})

File: backend/app/schemas/klimarelevanzpruefung_eingabe_fb3.py
from typing import Optional
from uuid import UUID
from pydantic import BaseModel, Field, ConfigDict


class KlimarelevanzpruefungEingabeFb3Base(BaseModel):
    """Required (non-nullable) fields"""

    c1q1: int
    c2q1: int


class KlimarelevanzpruefungEingabeFb3Optional(BaseModel):
    """Optional (nullable) fields"""

    # Teil 1
    c1q2: Optional[int] = None
    c1q3: Optional[str] = None
    c1q4: Optional[str] = None  # nullable=True, even if FK defined
    c1q5: Optional[str] = None
    c1q6: Optional[int] = None
    c1q7: Optional[str] = None
    c1q8: Optional[int] = None
    c1q9: Optional[str] = None

    # Teil 2
    c2q2: Optional[int] = None
    c2q3: Optional[str] = None
    c2q4: Optional[int] = None
    c2q5: Optional[str] = None
    c2q6: Optional[int] = None
    c2q7: Optional[str] = None
    c2q8: Optional[int] = None
    c2q9: Optional[str] = None


class KlimarelevanzpruefungEingabeFb3Create(
    KlimarelevanzpruefungEingabeFb3Base, KlimarelevanzpruefungEingabeFb3Optional
):
    """Schema for creating new entries"""

    pass


class KlimarelevanzpruefungEingabeFb3Update(KlimarelevanzpruefungEingabeFb3Optional):
    """Schema for updating entries (all fields optional)"""

    c1q1: Optional[int] = None
    c2q1: Optional[int] = None


class KlimarelevanzpruefungEingabeFb3Read(
    KlimarelevanzpruefungEingabeFb3Base, KlimarelevanzpruefungEingabeFb3Optional
):
    """Schema for reading entries"""

    model_config = ConfigDict(from_attributes=True)

    id: int = Field(..., description="Kennung für das Item")
